thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java

```diff
@@ -1131,4 +1165,29 @@ public Response getAlertFilterMTTD (@PathParam("functionId") @NotNull long id,
     double functionMTTDInHour = TimeUnit.HOURS.convert(detectionBucketSize, detectionUnit);
     return Response.ok(Math.max(functionMTTDInHour, alertFilterMTTDInHour)).build();
   }
+
+  /**
+   * Get Minimum Time to Detection for Autotuned (Preview) Alert Filter.
+   * This endpoint evaluate both alert filter's MTTD and bucket size for function and returns the maximum of the two as MTTD
+   * @param id autotune Id to be evaluated
+   * @param severity severity value
+   * @return minimum time to detection in HOUR
+   */
+  @GET
+  @Path("/eval/projected/mttd/{autotuneId}")
+  public Response getProjectedMTTD (@PathParam("autotuneId") @NotNull long id,
+      @QueryParam("severity") @NotNull double severity) {
+    //Initiate tuned alert filter
+    AutotuneConfigDTO target = DAO_REGISTRY.getAutotuneConfigDAO().findById(id);
+    Map<String, String> tunedParams = target.getConfiguration();
+    BaseAlertFilter alertFilter = alertFilterFactory.fromSpec(tunedParams);
+    // Get current function
+    long functionId = target.getFunctionId();
+    AnomalyFunctionDTO anomalyFunctionSpec = anomalyFunctionDAO.findById(functionId);
+    double alertFilterMTTDInHour = alertFilter.getAlertFilterMTTD(severity);
+    TimeUnit detectionUnit = anomalyFunctionSpec.getBucketUnit();
+    int detectionBucketSize = anomalyFunctionSpec.getBucketSize();
+    double functionMTTDInHour = TimeUnit.HOURS.convert(detectionBucketSize, detectionUnit);
+    return Response.ok(Math.max(functionMTTDInHour, alertFilterMTTDInHour)).build();
```

Comment:
Shouldn't this endpoint return a json string? The return of the endpoints in this file could be a json string or a plain string. Could you confirm which format is the one used in frontend?

Reply:
This endpoint will return one value of mttd, frontend can treat this return as a number which can directly display on UI.

Reply:
Yes, please @cecilynie : if you could make this method return a JSON string, it would make things alot smoother on the frontend. Thank you! Here's this for reference: https://github.com/linkedin/pinot/pull/2217